### PR TITLE
copr: add vectorized insert

### PR DESCRIPTION
**Pull request by trabbart**
Signed-off-by: Bartosz Podkanowicz <podkanowicz.bartosz@gmail.com>

### What problem does this PR solve?

vectorize function insert

### Check List #5751 

Tests 

- Unit test

### CODE REVIEW

**Comment by skyzh** [comment]
Hi @trabbart ! Recently I've been working on refactoring developer interface of `rpn_fn`. It seems that `insert` uses `Option<&Bytes>` as argument. In https://github.com/tikv/tikv/pull/8019 , we've changed that to `BytesRef`. If this PR gets merged before yours, you may have to make some changes. (Otherwise I'll fix that.) Thanks!

**Comment by trabbart** [comment]
Hi @skyzh ! Thanks for the information. Unfortunately after merge one test is failing. I don't know why.

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 1]
```diff
@@ -324,6 +324,38 @@ pub fn right(lhs: Option<BytesRef>, rhs: Option<&Int>) -> Result<Option<Bytes>>
...
+    len: Option<&Int>,
+    newstr: Option<BytesRef>,
+) -> Result<Option<Bytes>> {
+    match (s, pos, len, newstr) {
+        (Some(s), Some(pos), Some(len), Some(newstr)) => {
+            let pos = *pos;
+            let len = *len;
+            let upos: usize = pos as usize;
+            let mut ulen: usize = len as usize;
+            let s = s.to_vec();
```
Comment (iosmanthus) [inline components/tidb_query_vec_expr/src/impl_string.rs:341]: BytesRef's definition is `pub type BytesRef<'a> = &'a [u8];` which means we could eliminate this heap allocation here.

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 2]
```diff
@@ -324,6 +324,38 @@ pub fn right(lhs: Option<BytesRef>, rhs: Option<&Int>) -> Result<Option<Bytes>>
...
+            let newstr = newstr.to_vec();
```
Comment (iosmanthus) [inline components/tidb_query_vec_expr/src/impl_string.rs:342]: ditto

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 3]
```diff
@@ -324,6 +324,38 @@ pub fn right(lhs: Option<BytesRef>, rhs: Option<&Int>) -> Result<Option<Bytes>>
...
+    len: Option<&Int>,
+    newstr: Option<BytesRef>,
+) -> Result<Option<Bytes>> {
+    match (s, pos, len, newstr) {
+        (Some(s), Some(pos), Some(len), Some(newstr)) => {
+            let pos = *pos;
+            let len = *len;
+            let upos: usize = pos as usize;
+            let mut ulen: usize = len as usize;
+            let s = s;
```
Comment (zhongzc) [inline components/tidb_query_vec_expr/src/impl_string.rs:341]: I think it's better to eliminate these two redundant moves.

Comment (trabbart) [reply components/tidb_query_vec_expr/src/impl_string.rs]: I eliminated it.

**Comment by zhongzc** [comment]
@iosmanthus PTAL

**Pull request merged**